symfony/plugins/orangehrmSecurityAuthenticationPlugin/lib/service/SecurityAuthenticationService.php

```diff
@@ -0,0 +1,416 @@
+        }
+        return $this->authenticationService;
+    }
+
+    /**
+     *
+     * @param AuthenticationService $authService
+     */
+    public function setAuthenticationService($authService) {
+        $this->authenticationService = $authService;
+    }
+
+    /**
+     * @param $username
+     * @return array
+     * @throws ServiceException
+     */
+    public function searchForUserRecord($username) {
+
+
+        $result = $this->findUser($username);
+        list($user, $matchedByField, $matchedValue) = $result;
+
+        if ($user instanceof SystemUser) {
+            $associatedEmployee = $user->getEmployee();
+            if (!($associatedEmployee instanceof Employee)) {
+                throw new ServiceException(__('User account is not associated with an employee'));
+            } else {
+                $emailConfiguration=new EmailConfigurationService();
+                $companyEmail=$emailConfiguration->getEmailConfiguration();
+                if(!empty($companyEmail['sentAs'])) {
+                    if (empty($associatedEmployee['termination_id'])) {
+                        $workEmail = trim($associatedEmployee->getEmpWorkEmail());
+                        $employeeNumber = $associatedEmployee->getEmpNumber();
+                        if (empty($workEmail)) {
+                            if (empty($employeeNumber)) {
+                                throw new ServiceException(__('Please contact HR admin in order to reset the password'));
+                            } else {
+                                throw new ServiceException(__('Work Email Is Not Set. Please Contact HR Admin in Order to Reset the Password'));
+                            }
+                        }
+                    } else {
+                        throw new ServiceException(__('Please Contact HR Admin in Order to Reset the Password'));
+                    }
+                }else {
+                    throw new ServiceException(__('Password Reset Email Could Not Be Sent'));
+                }
+            }
+            if (!$this->hasPasswordResetRequest($workEmail)) {
+                return $result;
+            } else {
+                throw new ServiceException(__('There is a password reset request already in the system.'));
+            }
+        } elseif (empty($username)){
+            throw new ServiceException(__('Could not find a user with given details'));
+        } else {
+            throw new ServiceException(__('Please Contact HR Admin in Order to Reset the Password'));
+        }
+        return $result;
+    }
+
+    /**
+     * @param $username
+     * @return array
+     */
+    protected function findUser($username) {
+        $user = null;
+        $matchedByField = null;
+        $matchedValue = null;
+
+        $employeeService = $this->getEmployeeService();
+        $result = $employeeService->searchEmployee('emp_firstname', $username);
```

Comment:
How do you deal with terminated users?

Reply:
I guess the userService should be used instead of employeeService
```suggestion
        $result = $userService->searchSystemUsers(array('userName' => $username));
```